apps/src/applab/assetManagement/AssetUploader.jsx

```diff
@@ -0,0 +1,70 @@
+
+/**
+ * A component for managing hosted assets.
+ */
+module.exports = React.createClass({
+  propTypes: {
+    onUploadStart: React.PropTypes.func.isRequired,
+    onUploadDone: React.PropTypes.func.isRequired,
+    channelId: React.PropTypes.string.isRequired,
+    typeFilter: React.PropTypes.string,
+    uploadsEnabled: React.PropTypes.bool.isRequired
+  },
+
+  componentDidMount: function () {
+    var props = this.props;
+
+    $(React.findDOMNode(this.refs.uploader)).fileupload({
+      dataType: 'json',
+      url: '/v3/assets/' + props.channelId,
```

Comment:
https://en.wikipedia.org/wiki/Representational_state_transfer#Applied_to_web_services

to be RESTful, this should have a trailing slash. I know this can seem arbitrary, but sometimes these things come back to bite us, e.g. when we've tried to hook up AngularJS to a "REST" API that wasn't quite restful.